cmd/gitops/beta/run/cmd.go

```diff
@@ -548,6 +548,18 @@ func runCommandWithoutSession(cmd *cobra.Command, args []string) error {
 	// 2. log is the S3 logger that also delegates its outputs to "log0".
 	log0 := logger.NewCLILogger(os.Stdout)
 
+	state, err := term.GetState(int(os.Stdin.Fd()))
```

Comment:
Do you know exactly why this is necessary? I've never ran across terminal issues after exiting gitops run and I'd like us to be aligned on why the terminal state isn't as expected when exiting.

Reply:
I pressed Ctrl +C many times and was hit by the bug. It's not so hard to reproduce (but still a random behaviour).

Also, Paul Curtis reported it here https://github.com/weaveworks/weave-gitops/issues/3203

Reply:
Yes, I've seen it being reported but I'd like to understand the circumstances under which it occurs.